src/miniscript/astelem.rs

```diff
@@ -593,3 +595,233 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> Terminal<Pk, Ctx> {
         }
     }
 }
+
+impl<Ctx: ScriptContext> Terminal<DescriptorPublicKey, Ctx> {
+    /// Retrieve the assets associated with the type of miniscript element.
+    pub fn get_all_assets(&self) -> Vec<Assets> {
```

Comment:
Typically, in rust we don't name functions with `get_` prefix. Can you rename this to `all_assets`?